src/System.Linq.Parallel/tests/Combinatorial/ParallelQueryCombinationTests.cs

```diff
@@ -0,0 +1,1523 @@
+            Assert.All(ae.InnerExceptions, e => Assert.IsType<DeliberateTestException>(e));
+        }
+
+        [Theory]
+        [MemberData("UnaryOperators")]
+        [MemberData("BinaryOperators")]
+        public static void Except(LabelledOperation source, LabelledOperation operation)
+        {
+            int seen = defaultStart;
+            ParallelQuery<int> query = operation.Item(defaultStart, defaultSize + defaultSize / 2, source.Item)
+                .Except(operation.Item(defaultStart + defaultSize, defaultSize, source.Item));
+            foreach (int i in query)
+            {
+                Assert.Equal(seen++, i);
+            }
+            Assert.Equal(defaultStart + defaultSize, seen);
+        }
+
+        [Theory]
+        [MemberData("UnaryOperators")]
+        [MemberData("BinaryOperators")]
+        public static void Except_NotPipelined(LabelledOperation source, LabelledOperation operation)
+        {
+            int seen = defaultStart;
+            ParallelQuery<int> query = operation.Item(defaultStart, defaultSize + defaultSize / 2, source.Item)
+                .Except(operation.Item(defaultStart + defaultSize, defaultSize, source.Item));
+            Assert.All(query.ToList(), x => Assert.Equal(seen++, x));
+            Assert.Equal(defaultStart + defaultSize, seen);
+        }
+
+        [Theory]
+        [MemberData("UnaryOperators")]
+        [MemberData("BinaryOperators")]
+        public static void First(LabelledOperation source, LabelledOperation operation)
+        {
+            Assert.Equal(defaultStart, operation.Item(defaultStart, defaultSize, source.Item).First());
+        }
+
+        [Theory]
+        [MemberData("UnaryOperators")]
+        [MemberData("BinaryOperators")]
+        public static void First_Predicate(LabelledOperation source, LabelledOperation operation)
+        {
+            Assert.Equal(defaultStart + defaultSize / 2, operation.Item(defaultStart, defaultSize, source.Item).First(x => x >= defaultStart + defaultSize / 2));
+        }
+
+        [Theory]
+        [MemberData("UnaryOperators")]
+        [MemberData("BinaryOperators")]
+        public static void First_Predicate_None(LabelledOperation source, LabelledOperation operation)
+        {
+            Assert.Throws<InvalidOperationException>(() => operation.Item(defaultStart, defaultSize, source.Item).First(x => false));
+        }
+
+        [Theory]
+        [MemberData("UnaryFailingOperators")]
+        [MemberData("BinaryFailingOperators")]
+        public static void First_AggregateException(LabelledOperation source, LabelledOperation operation)
+        {
+            AggregateException ae = null;
+            // Concat seems able to return the first element when the left query does not fail ("first" query).
+            // This test might be flaky in the case that it decides to run the right query too...
```

Comment:
Sorry, what I was referring to was the fact that, per discussion after #584 was closed, I don't necessarily have a guarantee that `Concat` always calls the left query first (or before the right might start spooling), when unordered.  Or at least that's the way I read it at the time.